### ISSUE TITLE
Feature Request: Ability to configure some basic behaviors from the "fish_config" web UI

### ISSUE THREAD

**Issue by ideasman42** [fish-shell/fish-shell#3363]
While the web UI is already very useful, there are some basic behaviors new users may want to change, that are configurable but not very discoverable.

Suggesting a new section be added for *Behavior*, settings could include for example.

- [fish_prompt_pwd_dir_length](http://stackoverflow.com/questions/866989/fish-interactive-shell-full-path)
- [fish_greeting](http://stackoverflow.com/questions/13995857/suppress-or-customize-intro-message-in-fish-shell)
- __fish_git_prompt_show_informative_status (possibly others, but this is at least a way to toggle extra information).
-   [fish_vi_key_bindings](http://stackoverflow.com/questions/28444740/how-to-use-vi-mode-in-fish-shell)
Linked to people asking about how to perform these actions.

----
@Siecje  suggested I open an issue for this topic, on IRC.

**Comment by floam** [comment]
These would also be a bit more discoverable if we named them better. It would be nice to rename key variables to drop the `fish_` prefix to promote them from being buried aside stuff used mostly internally by our library of included scripts. Stuff like `fish_vi_key_bindings` manes no sense to me - it ought too just be `$vi_key_bindings`. It's fish's namespace to use.

**Comment by krader1961** [comment]
> It's fish's namespace to use.

Yes, and no. Mostly no because there are no namespaces outside of block scoping of vars (which isn't really the same thing). We don't have to employ guidelines such as prefixing fish specific variables with `fish_` and internal use only vars with `__fish_` but doing so helps avoid problems. At least to the extent we are consistent about doing so and clearly communicate such guidelines to the fish community. I'll grant you we have not done a great job on either of those points. But I fail to see how abandoning the guideline makes the situation better.

**Comment by floam** [comment]
I'm not saying abandon the guideline for all of them. Just the ones that are intended to be directly used by the end-user.

**Comment by floam** [comment]
The list at the top of this topic is a good one. Another that sticks out to me would be `fish_title` - which is as "built-in" not in the `builtin` sense something could be. It might have better been named `title` or `terminal_title`. A problem is when you type `fish_` and hit tab the signal to noise is terrible - by not having `fish_`, they'd actually be much easier to find.

**Comment by ideasman42** [comment]
Shouldn't naming of fish options be a separate issue?
*(these kinds of naming cleanups may need to be done from time to time, and don't relate to adding options to the web UI).*

**Comment by floam** [comment]
You're right.

**Comment by ideasman42** [comment]
This is tagged as needing docs, whats this mean practically?

**Comment by faho** [comment]
@ideasman42: That means that properly solving this means adding something to the docs. If we are to add a new tab to fish_config, we'll probably want to have a mention of that in the documentation.
